transport/grpc/options.go

```diff
@@ -83,6 +84,17 @@ func Logger(logger *zap.Logger) TransportOption {
 	}
 }
 
+// RequestValidator specifies an option to validate a transport.Request
+// before allowing an outbound call to be made or an inbound call
+// to be processed.
+//
+// By default, no validation is done.
+func RequestValidator(requestValidator func(*transport.Request) error) TransportOption {
```

Comment:
Just a thought: If you can see the case for multiple request validators, we could maybe make them composable.

Reply:
Made it possible to use this option multiple times.